cloudinit/cmd/main.py

```diff
@@ -328,9 +328,12 @@ def main_init(name, args):
     outfmt = None
     errfmt = None
     try:
-        early_logs.append((logging.DEBUG, "Closing stdin."))
-        util.close_stdin()
-        (outfmt, errfmt) = util.fixup_output(init.cfg, name)
+        if not os.isatty(sys.stdin.fileno()):
+            early_logs.append((logging.DEBUG, "Closing stdin."))
+            util.close_stdin()
+        else:
+            LOG.warning("Not closing stdin, stdin is a tty.")
```

Comment:
Is there a reason for this to be a warning? I think it makes more sense as a debug.